src/serial-protocol.c

```diff
@@ -27,6 +28,7 @@
 #define ESC_BYTE 0x1B
 
 #define DEFAULT_BUFFER_SIZE 1024
+#define DEFAULT_MAXIMUM_DECODER_BUFFER_SIZE (1 * 1024 * 1024) /* 1 MB */
```

Comment:
Why not ???
The maximum here is here to limit the decoder capacity to avoid to allocate too much memory in case of malformed incoming frame. The default value is quite arbitrary and may be increased in the future.